File: BackendAPI/Application/DTOs/Output/KwekerStatsOutputDto.cs
namespace Application.DTOs.Output;

public class KwekerStatsOutputDto
{
    public required int TotalProducts { get; set; }
    public required int ActiveAuctions { get; set; }
    public required decimal TotalRevenue { get; set; }
    public required int OrdersReceived { get; set; }
    public required List<MonthlyRevenueDto> MonthlyRevenue { get; set; }
    public required List<DailyRevenueDto> DailyRevenue { get; set; }
}

public class MonthlyRevenueDto
{
    public required int Year { get; set; }
    public required int Month { get; set; }
    public required string MonthName { get; set; }
    public required decimal Revenue { get; set; }
}

public class DailyRevenueDto
{
    public required int Year { get; set; }
    public required int Month { get; set; }
    public required int Day { get; set; }
    public required string DateLabel { get; set; }
    public required decimal Revenue { get; set; }
}

File: FrontendWeb/src/declarations/dtos/output/KwekerStatsOutputDto.ts
// KwekerStatsOutputDto.ts
export interface KwekerStatsOutputDto {
	totalProducts: number;
	activeAuctions: number;
	totalRevenue: number;
	ordersReceived: number;
	monthlyRevenue: MonthlyRevenueDto[];
	dailyRevenue: DailyRevenueDto[];
}

export interface MonthlyRevenueDto {
	year: number;
	month: number;
	monthName: string;
	revenue: number;
}

export interface DailyRevenueDto {
	year: number;
	month: number;
	day: number;
	dateLabel: string;
	revenue: number;
}
